EIPS/eip-8032.md

```diff
@@ -29,59 +31,153 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
 |Name|Value|Description|
 |-|:-:|-|
 |`FORK_TIMESTAMP`| TBD | Fork activation timestamp |
-|`EXP_FACTOR` | TBD | Exponent gas cost factor |
 |`LIN_FACTOR` | TBD | Linear gas cost factor |
 | `ACTIVATION_THRESHOLD` | TBD | Activation threshold, chosen to be at ~8GB of data. |
+| `TRANSITION_REGISTRY_ADDRESS` | TBD | The address of system contract that holds the information of account transition |
+| `TRANSITION_SLOTS_PER_BLOCK` | TBD | The number of storage slots to iterate per block during the transition process |
 
-### Account tree and update rules change
 
-Account RLP descriptors receive an optional `depth` field, corresponding to the maximum depth written to since `FORK_TIMESTAMP`.
+### New account field
 
-After block execution and while computing the state root for the account, take note of the maximum depth written to.
+Account RLP descriptors receive an optional `storage_count` field, corresponding to the current number of storage slots it contains.
 
-While updating the account's root hash, also update its `depth` fields according to the following rules:
+### Account transition
 
-```python=
-newroot, max_depth = compute_root_and_max_depth()
-if account.depth < max_depth:
-    account.depth += 1
-    
-account.root = new_root
-```
+At `FORK_TIMESTAMP`, the transition is activated.
+
+The transition process involves reading all accounts with non-empty code hashes and count the number of storage slots, which in turns populate the `storage_count` field in the account.
+
+The progress of the transition is stored at system contract at `TRANSITION_REGISTRY_ADDRESS`.
+
+Storage layout at `TRANSITION_REGISTRY_ADDRESS`:
+- slot `0x00`: `cursor_account_hash` (32-byte account hash)
+- slot `0x01`: `cursor_slot_hash` (32-byte slot hash last processed, or 0x00 if none)
```

Comment:
that's a bit of a problem btw, the semantics should be: next slot to be processed, because 0 is a valid hash. So if you reach the end of an account, this should be set back to 0 as the first hash of the next account.